.gitlab-ci/terraform.yml

```diff
@@ -28,6 +28,15 @@
     - terraform validate -var-file=cluster.tfvars contrib/terraform/$PROVIDER
     - terraform fmt -check -diff contrib/terraform/$PROVIDER
 
+.terraform_gcpvalidate:
+  extends: .terraform_install
+  stage: unit-tests
+  tags: [light]
+  only: ['master', /^pr-.*$/]
+  script:
+    - cd /builds/kargo-ci/kubernetes-sigs-kubespray/contrib/terraform/$PROVIDER && terraform validate -var-file=cluster.tfvars
+    - cd /builds/kargo-ci/kubernetes-sigs-kubespray/contrib/terraform/$PROVIDER && terraform fmt -check -diff 
+
```

Comment:
There is no reason to make an exception for GCP. The same code works for aws, openstack and packet. You need to change the code here: 
```
  on contrib/terraform/gcp/kubernetes.tf line 8, in provider "google":
   8:   credentials = file("${path.cwd}/clustertfiles/gcp-auth-config-json/account.json")
```

And make that configurable through a TF variable rather than a hardcoded path. According to [this](https://www.terraform.io/docs/providers/google/guides/provider_reference.html#credentials) `crendentials` can also take the JSON content instead of the file path, that could help you.

Reply:
As per the terraform documentation, `credentials` Either the path to or the contents of a service account key file in JSON format. Which i am already using it, as per your comments, i will take `service-account key file` path TF variable.